### PR TITLE
Separate common and persistence libs

### PR DESCRIPTION
**Pull request by aug24**
## What does this change?
More work on separation of concerns.  This PR pulls the elasticsearch persistence layer into its own library.

## How can success be measured?
No changes to behaviour, admin lambda no longer needs exclusion of elasticsearch jars, which renders the build safer.

## Tested?
- [ ] locally by committer
- [ ] locally by Guardian reviewer
- [ ] on the Guardian's TEST environment

### CODE REVIEW

**Review by sihil — COMMENTED** [review]
Couple of thoughts...

File: build.sbt [hunk 1]
```diff
@@ -120,6 +117,18 @@ lazy val restLib = project("rest-lib").settings(
     ws,
   ),
 
+  dependencyOverrides += "org.apache.thrift" % "libthrift" % "0.9.1"
```
[the remark this answers is not in the document]
Comment (aug24) [reply build.sbt:120]: Originally line 117, at the end of the section defining common-lib.

File: build.sbt [hunk 2]
```diff
@@ -281,7 +286,7 @@ val buildInfo = Seq(
 def playProject(projectName: String, port: Int, path: Option[String] = None): Project =
   project(projectName, path)
     .enablePlugins(PlayScala, JDebPackaging, SystemdPlugin, BuildInfoPlugin)
-    .dependsOn(restLib)
+    .dependsOn(restLib, persistenceLib)
```
[the remark this answers is not in the document]
Comment (sihil) [reply build.sbt:284]: Think this still stands?

[the remark this answers is not in the document]
Comment (aug24) [reply build.sbt:284]: Yes: This is the ElasticSearch persistence only.  Perhaps that needs to be made more clear.

**Pull request closed without merging**